pyrdp/parser/rdp/negotiation.py

```diff
@@ -93,8 +93,13 @@ def parse(self, data: bytes) -> NegotiationResponsePDU:
             type = Uint8.unpack(stream)
             flags = Uint8.unpack(stream)
             length = Uint16LE.unpack(stream)
-            selectedProtocols = Uint32LE.unpack(stream)
-            return NegotiationResponsePDU(type, flags, selectedProtocols)
+
+            if type == NegotiationType.TYPE_RDP_NEG_FAILURE:
+                failureCode = Uint32LE.unpack(stream)
+                return NegotiationFailurePDU(type, flags, failureCode)
+            else:
+                selectedProtocols = Uint32LE.unpack(stream)
+                return NegotiationResponsePDU(type, flags, selectedProtocols)
```

Comment:
As I said in my previous comment, I think this should be changed because the return type isnt telling the truth now. Since the only difference is that selectedProtocol == failureCode for the FailurePDU, maybe make `NegotiationFailurePDU` inherit `NegotiationResponsePDU` and add something like `failureCode = selectedProtocols` in the constructor? @xshill what do you think?

Reply:
Its a bad idea